src/lib/dialog_ui.sh

```diff
@@ -377,22 +380,41 @@ function create_choice_list_screen()
   local message_box="$2"
   local -n _choices="$3"
   local -n _check_statuses="$4"
-  local height="$5"
-  local width="$6"
-  local flag="$7"
+  local ok_label="$5"
+  local cancel_label="$6"
+  local extra_button_label="$7"
+  local height="$8"
+  local width="$9"
+  local flag="${10}"
   local choice_description
   local index=0
   local cmd
 
   flag=${flag:-'SILENT'}
   height=${height:-$DEFAULT_HEIGHT}
   width=${width:-$DEFAULT_WIDTH}
+  cancel_label=${cancel_label:-'Cancel'}
+  ok_label=${ok_label:-'Ok'}
+
+  # Escape all single quotes to avoid breaking arguments
+  ok_label=$(str_escape_single_quotes "$ok_label")
+  cancel_label=$(str_escape_single_quotes "$cancel_label")
+  extra_button_label=$(str_escape_single_quotes "$extra_button_label")
 
   # Escape all single quotes to avoid breaking arguments
   box_title=$(str_escape_single_quotes "$box_title")
   message_box=$(str_escape_single_quotes "$message_box")
 
   cmd=$(build_dialog_command_preamble "$box_title")
+
```

Comment:
Also a nitpick. The block of code that builds the `cmd` variable is contiguos in the "pattern". Can you remove this blank line?